src/agent/onefuzz-agent/src/tasks/regression/regression.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::tasks::{
+    heartbeat::*,
+    report::crash_report::{CrashReport, CrashTestResult},
+    utils::download_input,
+};
+use anyhow::Result;
+use async_trait::async_trait;
+use onefuzz::syncdir::SyncedDir;
+use reqwest::Url;
+use std::path::PathBuf;
+
+#[async_trait]
+pub trait RegressionHandler {
+    async fn get_crash_result(
+        &self,
+        input: PathBuf,
+        input_url: Option<Url>,
+    ) -> Result<CrashTestResult>;
+    async fn save_regression(
+        &self,
+        crash_result: CrashTestResult,
+        original_report: Option<CrashReport>,
+    ) -> Result<()>;
+}
+
+pub async fn run(
+    heartbeat_client: Option<TaskHeartbeatClient>,
+    input_reports: &Option<SyncedDir>,
+    crashes: &Option<SyncedDir>,
+    inputs: &Option<SyncedDir>,
+    report_list: &[String],
```

Comment:
I think we need to document what a `report_list` is, and anything we're assuming about it.